File: React-Next with DotNet-CosmosDB/server/Basketball/Microservice/ViewModels/SeasonVM.cs
using NBA.Infrastructure;
using NBA.Models;

namespace NBA.ViewModels
{
    internal class SeasonVM : BasketballViewModel
    {
        public string Id { get; set; }

        public override (bool Success, string Message) Validate()
        {
            return (false, "this viewModel cannot be used for write operations");
        }

        internal static SeasonVM FactorFrom(Season model)
        {
            return new() { Id = model.Id};
        }
    }
}

File: Vue-Nuxt with Java-MongoDB/server/Basketball/Microservice/src/main/java/com/nba/microservice/viewmodels/SeasonVM.java
package com.nba.microservice.viewmodels;

import com.nba.microservice.models.Season;
import com.nba.microservice.infrastructure.BasketballViewModel;
import com.nba.microservice.infrastructure.ValidationResult;

public class SeasonVM extends BasketballViewModel {

    private String id;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    @Override
    public ValidationResult validate() {
        return new ValidationResult(false, "this viewModel cannot be used for write operations");
    }

    public static SeasonVM factorFrom(Season model) {
        SeasonVM seasonVM = new SeasonVM();
        seasonVM.setId(model.getId());
        return seasonVM;
    }
}